apps/web/src/views/BuyCrypto/components/OnRampCurrencySelect/index.tsx

```diff
@@ -0,0 +1,183 @@
+  border: 1px solid ${({ theme, error }) => (error ? theme.colors.failure : theme.colors.inputSecondary)};
+  border-radius: 16px;
+  background: ${({ theme }) => theme.colors.input};
+  cursor: pointer;
+  position: relative;
+  min-width: 136px;
+  user-select: none;
+  z-index: 20;
+
+  ${({ theme }) => theme.mediaQueries.sm} {
+    min-width: 168px;
+  }
+`
+
+const CurrencySelectButton = styled(Button).attrs({ variant: 'text', scale: 'sm' })`
+  padding: 0px;
+  border-left: ${({ theme }) => `1px solid ${theme.colors.inputSecondary}`};
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+  width: 130px;
+  padding-left: 16px;
+  padding-right: 8px;
+  border-radius: 0px;
+`
+
+const ButtonAsset = ({
+  id,
+  selectedCurrency,
+  currencyLoading,
+}: {
+  id: string
+  selectedCurrency: Currency
+  currencyLoading: boolean
+}) => {
+  const { t } = useTranslation()
+  return (
+    <Flex pl="8px">
+      {id === 'onramp-input' ? (
+        <FiatLogo currency={selectedCurrency} size="24px" style={{ marginRight: '8px' }} />
+      ) : (
+        <CurrencyLogo currency={selectedCurrency} size="24px" style={{ marginRight: '8px' }} />
+      )}
+      {currencyLoading ? null : (
+        <Text id="pair" bold>
+          {(selectedCurrency && selectedCurrency.symbol && selectedCurrency.symbol.length > 10
+            ? `${selectedCurrency.symbol.slice(0, 4)}...${selectedCurrency.symbol.slice(
+                selectedCurrency.symbol.length - 5,
+                selectedCurrency.symbol.length,
+              )}`
+            : selectedCurrency?.symbol) || t('Select a currency')}
+        </Text>
+      )}
+    </Flex>
+  )
+}
+
+interface CurrencySelectProps extends CurrencySearchModalProps, BoxProps {
+  id: 'onramp-input' | 'onramp-output'
+  onUserInput?: (value: string) => void
+  value?: string
+  disableCurrencySelect?: boolean
+  currencyLoading?: boolean
+  error?: boolean
+  errorText?: string
+  onInputBlur?: () => void
+  disabled?: boolean
+  loading?: boolean
+  topElement?: ReactNode
+  bottomElement?: ReactNode
+}
+
+export const CurrencySelect = ({
+  onCurrencySelect,
+  onUserInput,
+  onInputBlur,
+  selectedCurrency,
+  otherSelectedCurrency,
+  showCommonBases,
+  commonBasesType,
+  id,
+  currencyLoading,
+  topElement,
+  error,
+  value,
+  bottomElement,
+  ...props
+}: CurrencySelectProps) => {
+  const onRampTokens = useAllOnRampTokens()
+  const tokensToShow = id === 'onramp-input' ? fiatCurrencyMap : (onRampTokens as any)
+
+  const [onPresentCurrencyModal] = useModal(
+    <CurrencySearchModal
+      onCurrencySelect={onCurrencySelect}
+      selectedCurrency={selectedCurrency}
+      otherSelectedCurrency={otherSelectedCurrency}
+      showCommonBases={showCommonBases}
+      commonBasesType={commonBasesType}
+      tokensToShow={tokensToShow}
+      mode={id}
+    />,
+  )
+
+  const handleUserInput = useCallback(
+    (val: string) => {
+      onUserInput(val)
```

Comment:
If there's no extra logic, we can just use the `onUserInput`

Reply:
resolved